ci/cloudbuild/builds/gcs-grpc.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+#
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+source "$(dirname "$0")/../../lib/init.sh"
+source module ci/cloudbuild/builds/lib/bazel.sh
+source module ci/cloudbuild/builds/lib/integration.sh
+
+export CC=clang
+export CXX=clang++
+
+mapfile -t args < <(bazel::common_args)
+bazel test "${args[@]}" --test_tag_filters=-integration-test ...
+
+mapfile -t integration_args < <(integration::args)
+GOOGLE_CLOUD_CPP_STORAGE_GRPC_CONFIG=media # Uses the gRPC data plane
+integration::bazel_with_emulators test "${args[@]}" "${integration_args[@]}"
```

Comment:
Soon (~next week) we would want to run the storage integration tests against production.  I assume you would be Okay if we changed `integration::bazel_with_emulators` to support switching one service to production?

Right now we are using the emulator because the GCS+gRPC service was down for about a month, but it is back up as of this week.

Reply:
> Soon (~next week) we would want to run the storage integration tests against production. I assume you would be Okay if we changed `integration::bazel_with_emulators` to support switching one service to production?

I just changed this test script to only run the GCS integration tests, which seems reasonable since the purpose of this build is testing GCS + gRPC.

Regarding changing to test against production: Would that be a change in `google/cloud/storage/ci/run_integration_tests_emulator_bazel.sh` or in this build script (`ci/cloudbuild/builds/gcs-grpc.sh`)? Either way, a change in either of those to hit production instead would be fine.

> 
> Right now we are using the emulator because the GCS+gRPC service was down for about a month, but it is back up as of this week.

Should I make the change to hit production right now? How would I do that?

Reply:
> Should I make the change to hit production right now? How would I do that?

You need to revert the changes in #6063, it would make my life easier if you would. But it is more involved than adding a build.

Reply:
> > Should I make the change to hit production right now? How would I do that?
> 
> You need to revert the changes in #6063, it would make my life easier if you would. But it is more involved than adding a build.

I took a stab at doing this, but ended up with a few extra errors that weren't obvious. It doesn't look difficult to enable, but it's probably best done in a separate, focused PR. I think this whole script would amount to something like:

```shell
mapfile -t args < <(bazel::common_args)
mapfile -t integration_args < <(integration::args)
GOOGLE_CLOUD_CPP_STORAGE_GRPC_CONFIG=media # Uses the gRPC data plane
bazel test "${args[@]}" "${integration_args[@]}" google/cloud/storage/...
```
--
That said, is the PR as-is a reasonable replacement for the kokoro build we have today named `gcs-grpc`, or is it missing more stuff?